### ISSUE TITLE
SSL Codes

### ISSUE THREAD

**Issue by khairulislam** [chaudatascience/cha_mae_vit#2]
Hi,

Thank you for sharing the codes. I was reproducing the pretraining, but I found that only the ChaMAEViT code was available. For example, the paper mentions SimCLR to DINOv2 for comparison, but their code isn't available here. 

<img width="912" height="366" alt="Image" src="https://github.com/user-attachments/assets/76e4e7f0-4b6e-4ecc-b3ab-607deba3e2dc" />

**Comment by khairulislam** [comment]
Also, did you try pretraining with SSL and then fine-tuning? Currently, the training objective of ChA-MAEViT looks hybrid; it optimizes SSL reconstruction loss and task loss at the same time. TIA.

**Comment by chaudatascience** [comment]
Hi,

Thanks for being interested in our work!

1/ DiChaViT + SSL in Table 2.

In Table 2, we combine [DiChaViT](https://github.com/chaudatascience/diverse_channel_vit/blob/main/models/dichavit.py) with an SSL loss and jointly optimize both the SSL and task losses (i.e., we didn't pretrain on SSL and then fine-tune on the task loss; both losses were optimized simultaneously).

We mainly used the code from their papers with small modifications. For example:
DiChaViT+SimCLR, we used the SimCLR loss from [here](https://github.com/sthalles/SimCLR/blob/master/simclr.py). You can integrate that into the current code: Only changes are adding SimCLR loss, and using SimCLR augmentation for the dataset.
For DiChaViT+DINOv2, we use the code from [their paper](https://github.com/facebookresearch/dinov2). To get the result, you can just swap out the vit backbone by [DiChaViT backbone](https://github.com/chaudatascience/diverse_channel_vit/blob/main/models/dichavit.py).
For DiChaViT+MAE, we used the model from [here](https://github.com/recursionpharma/maes_microscopy)

2/ Pretraining with SSL and then fine-tuning?
No, as mentioned above, we optimized both objectives at the same time, no pretraining. You can find more details in [Appendix C: Combination of DiChaViT with SSL](https://arxiv.org/pdf/2503.19331).

3/ ChA-MAEViT's training objective
Yes. When the label is available, we optimized both task loss and MAE loss at the same time.
Most of our experiments use both self-supervised (SSL) and supervised learning to boost performance. That said, in Section F, we also evaluate SSL by itself.

Hope these help!

Best,
Chau

**Comment by khairulislam** [comment]
Thanks for your prompt and detailed response. That addresses my question.
